src/hackerrank-problems/camel-case4.ts

```diff
@@ -0,0 +1,88 @@
+/*
+    Problem:
+    Camel Case is a naming style common in many programming languages. In Java, 
+    method and variable names typically start with a lowercase letter, with all 
+    subsequent words starting with a capital letter (example: startThread). Names 
+    of classes follow the same pattern, except that they start with a capital 
+    letter (example: BlueCar).
+
+    Your task is to write a program that creates or splits Camel Case variable,
+    method, and class names.
+
+    Important note:
+    The following code is copy pasted from HackerRank and runs only in their
+    environment.
+*/
+'use strict';
+
+process.stdin.resume();
+process.stdin.setEncoding('utf-8');
+let inputString: string = '';
+let inputLines: string[] = [];
+let currentLine: number = 0;
+process.stdin.on('data', function(inputStdin: string): void {
+    inputString += inputStdin;
+});
+
+process.stdin.on('end', function(): void {
+    inputLines = inputString.split('\n');
+    inputString = '';
+    main();
+});
+
+function readLine(): string {
+    return inputLines[currentLine++];
+}
+
+function main() {
+    // Enter your code here
+    // Read the input
+    while (currentLine < inputLines.length) {
+        // Replace carriage return with empty string
+        const input = inputLines[currentLine].replace(/\r/, '')
+        // Split 
+        if (input.startsWith('S')) {
+            console.log(splitCamelCase(input))
+        }
+        // Combine
+        else {
+            console.log(combineCamelCase(input))
+        }
+        currentLine++
+    }
+}
```

Comment:
This types of comments are not adding value here, I recommend to use comments just on pure utility functions. And not in code bit above function signature.

Reply:
Ok, got it - I removed the unnecessary comments.